File: mediapicker/src/main/java/vn/tungdx/mediapicker/widget/PickerImageView.java
package vn.tungdx.mediapicker.widget;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.util.AttributeSet;
import android.widget.ImageView;

import vn.tungdx.mediapicker.R;


/**
 * @author TUNGDX
 */

/**
 * Display thumbnail of video, photo and state when video, photo selected or
 * not.
 */
public class PickerImageView extends ImageView {
    private Paint paintBorder;

    private boolean isSelected;
    private int borderSize = 1;

    public PickerImageView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        init();
    }

    public PickerImageView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    private void init() {
        paintBorder = new Paint();
        paintBorder.setAntiAlias(true);
        paintBorder.setColor(getResources().getColor(R.color.picker_color));
        borderSize = getResources().getDimensionPixelSize(
                R.dimen.picker_border_size);
    }

    public PickerImageView(Context context) {
        super(context);
        init();
    }

    public void setSelected(boolean isSelected) {
        if (isSelected != this.isSelected) {
            this.isSelected = isSelected;
            invalidate();
        }
    }

    public boolean isSelected() {
        return isSelected;
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        int width = getMeasuredWidth();
        setMeasuredDimension(width, width);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        if (isSelected) {
            canvas.drawRect(0, 0, borderSize, getHeight(), paintBorder);
            canvas.drawRect(getWidth() - borderSize, 0, getWidth(),
                    getHeight(), paintBorder);
            canvas.drawRect(0, 0, getWidth(), borderSize, paintBorder);
            canvas.drawRect(0, getHeight() - borderSize, getWidth(),
                    getHeight(), paintBorder);
        }
    }
}

File: mediapicker/src/main/java/vn/tungdx/mediapicker/widget/PickerImageView.kt
package vn.tungdx.mediapicker.widget

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.util.AttributeSet
import android.widget.ImageView

import vn.tungdx.mediapicker.R


/**
 * @author TUNGDX
 */

/**
 * Display thumbnail of video, photo and state when video, photo selected or
 * not.
 */
class PickerImageView : ImageView {
    private var paintBorder: Paint? = null

    private var isSelected: Boolean = false
    private var borderSize = 1

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init()
    }

    private fun init() {
        paintBorder = Paint()
        paintBorder!!.isAntiAlias = true
        paintBorder!!.color = resources.getColor(R.color.picker_color)
        borderSize = resources.getDimensionPixelSize(
                R.dimen.picker_border_size)
    }

    constructor(context: Context) : super(context) {
        init()
    }

    override fun setSelected(isSelected: Boolean) {
        if (isSelected != this.isSelected) {
            this.isSelected = isSelected
            invalidate()
        }
    }

    override fun isSelected(): Boolean {
        return isSelected
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        val width = measuredWidth
        setMeasuredDimension(width, width)
    }

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        if (isSelected) {
            canvas.drawRect(0f, 0f, borderSize.toFloat(), height.toFloat(), paintBorder!!)
            canvas.drawRect((width - borderSize).toFloat(), 0f, width.toFloat(),
                    height.toFloat(), paintBorder!!)
            canvas.drawRect(0f, 0f, width.toFloat(), borderSize.toFloat(), paintBorder!!)
            canvas.drawRect(0f, (height - borderSize).toFloat(), width.toFloat(),
                    height.toFloat(), paintBorder!!)
        }
    }
}